### PR TITLE
add an option to allow removal of the -Dusethreads flag in Perl easyblock

### PR DESCRIPTION
**Pull request by valtandor**
Add an option to allow removal of the -Dusethreads flag

### CODE REVIEW

**Comment by valtandor** [comment]
Yes.

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyblocks/p/perl.py [hunk 1]
```diff
@@ -42,17 +43,27 @@
...
         """
         configopts = ' '.join([
             self.cfg['configopts'],
-            "-Dusethreads",
             '-Dcc="%s"' % os.getenv('CC'),
             '-Dccflags="%s"' % os.getenv('CFLAGS'),
             '-Dinc_version_list=none',
         ])
+        if self.cfg['use_perl_threads']:
+            configopts.append("-Dusethreads")
```
Comment (boegel) [inline easybuild/easyblocks/p/perl.py]: `configopts` is never a list, so `.append` will not work...

File: easybuild/easyblocks/p/perl.py [hunk 2]
```diff
@@ -42,17 +43,27 @@
...
+
         cmd = './Configure -de %s -Dprefix="%s" ' % (configopts, self.installdir)
```
Comment (boegel) [inline easybuild/easyblocks/p/perl.py]: maybe use `' '.join(configopts)` here instead of `configopts`, and remove it above (then the `.append` will work)

**Comment by valtandor** [comment]
@boegel: I've replaced the append with a join, by analogy with the preceding block. I hope this addresses the problem.

File: easybuild/easyblocks/p/perl.py [hunk 3]
```diff
@@ -42,17 +43,30 @@
...
             '-Dccflags="%s"' % os.getenv('CFLAGS'),
             '-Dinc_version_list=none',
         ])
+        if self.cfg['use_perl_threads']:
+            configopts = ' '.join([
+                self.cfg['configopts'],
+                '-Dusethreads',
+            ])
+
         cmd = './Configure -de %s -Dprefix="%s" ' % (configopts, self.installdir)
```
Comment (boegel) [inline easybuild/easyblocks/p/perl.py]: @valtandor: I would clean this up a bit as follows, i.e. construct `configopts` as a list first, and only `join` when it's used:

```python
configopts = [
...
]
if self.cfg['use_perl_threads']:
    configopts.append('-Dusethreads')

cmd = './Configure -de %s -Dprefix="%s" ' % (' '.join(configopts), self.installdir)
```

**Comment by valtandor** [comment]
@boegel: Done.

File: easybuild/easyblocks/p/perl.py [hunk 4]
```diff
@@ -42,18 +43,27 @@
...
-            '-Dcc="%s"' % os.getenv('CC'),
-            '-Dccflags="%s"' % os.getenv('CFLAGS'),
-            '-Dinc_version_list=none',
-        ])
-        cmd = './Configure -de %s -Dprefix="%s" ' % (configopts, self.installdir)
+        configopts = []
+        configopts.append(self.cfg['configopts'])
+        configopts.append('-Dcc="{0}"'.format(os.getenv('CC')))
+        configopts.append('-Dccflags="{0}"'.format(os.getenv('CFLAGS')))
+        configopts.append('-Dinc_version_list=none')
```
Comment (boegel) [inline easybuild/easyblocks/p/perl.py]: I didn't mean to imply you should use `.append` all over the place... just define the initial `configopts` in one go, and append `-Dusethreads` if desired

see https://github.com/valtandor/easybuild-easyblocks/pull/1

**Comment by boegel** [comment]
Going in, thanks @valtandor!

**Pull request merged**